internal/satellite/module/gatherer/receiver_gatherer.go

```diff
@@ -41,51 +44,75 @@ type ReceiverGatherer struct {
 
 	// self components
 	outputChannel chan *queue.SequenceEvent
+	// metrics
+	receiveCounter     *prometheus.CounterVec
+	queueOutputCounter *prometheus.CounterVec
 }
 
 func (r *ReceiverGatherer) Prepare() error {
-	log.Logger.Infof("receiver gatherer module of %s namespace is preparing", r.config.NamespaceName)
-	r.runningReceiver.RegisterHandler(r.runningServer)
+	log.Logger.Infof("receiver gatherer module of %s namespace is preparing", r.config.PipeName)
+	r.runningReceiver.RegisterHandler(r.runningServer.GetServer())
 	if err := r.runningQueue.Initialize(); err != nil {
-		log.Logger.Infof("the %s queue of %s namespace was failed to initialize", r.runningQueue.Name(), r.config.NamespaceName)
+		log.Logger.Infof("the %s queue of %s namespace was failed to initialize", r.runningQueue.Name(), r.config.PipeName)
 		return err
 	}
+	r.receiveCounter = prometheus.NewCounterVec(prometheus.CounterOpts{
```

Comment:
I am not very clear about this. Should the metrics here be defined in plugins ? Or we supports prometheus in `Gather` by default?

Reply:
the feature is to monitor self status.  Could you explain that again? I'm not very clear

Reply:
> the feature is to monitor self status

OK, This is the metric for satellite self.
I mistakenly thought that this is an extension of Prometheus Gather.

Reply:
@EvanLjp I think you should rename this PR to `Satellite Self-Observability` :) I used to think this is for scratch from prometheus endpoint.